packages/protocol/contracts-0.8/common/MintGoldSchedule.sol

```diff
@@ -0,0 +1,315 @@
+    require(
+      FixidityLib.fixed1().gte(communityRewardFraction.add(carbonOffsettingFraction)),
+      "Sum of partner fractions must be less than or equal to 1."
+    );
+    emit CommunityRewardFractionSet(value);
+    return true;
+  }
+
+  /**
+   * @notice Sets the carbon offsetting fund.
+   * @param partner The address of the carbon offsetting partner.
+   * @param value The percentage of the total reward to be sent to the carbon offsetting partner as Fixidity fraction.
+   * @return True upon success.
+   */
+  function setCarbonOffsettingFund(
+    address partner,
+    uint256 value
+  ) public onlyOwner whenDependenciesSet returns (bool) {
+    require(partner != address(0), "Partner cannot be the zero address.");
+    uint256 timeSinceL2Start = block.timestamp - l2StartTime;
+    uint256 totalL2LinearSecondsAvailable = SECONDS_LINEAR - (l2StartTime - genesisStartTime);
+    require(
+      timeSinceL2Start < totalL2LinearSecondsAvailable,
+      "Can only update fraction once block reward calculation for years 15-30 has been implemented."
+    );
+    FixidityLib.Fraction memory wrappedValue = FixidityLib.wrap(value);
+    require(
+      partner != carbonOffsettingPartner || !wrappedValue.equals(carbonOffsettingFraction),
+      "Partner and value must be different from existing carbon offsetting fund."
+    );
+    require(wrappedValue.lt(FixidityLib.fixed1()), "Value must be less than 1.");
+    carbonOffsettingPartner = partner;
+    carbonOffsettingFraction = wrappedValue;
+    require(
+      FixidityLib.fixed1().gte(communityRewardFraction.add(carbonOffsettingFraction)),
+      "Sum of partner fractions must be less than or equal to 1."
+    );
+    emit CarbonOffsettingFundSet(partner, value);
+    return true;
+  }
+
+  /**
+   * @notice Calculates remaining CELO balance to mint.
+   * @return The remaining CELO balance to mint.
+   */
+  function getRemainingBalanceToMint() public view returns (uint256) {
+    return GOLD_SUPPLY_CAP - getGoldToken().totalSupply();
+  }
+
+  /**
+   * @return The total balance minted by the MintGoldSchedule contract.
+   */
+  function getTotalMintedBySchedule() public view returns (uint256) {
+    return totalMintedBySchedule;
+  }
+
+  /**
+   * @return The currently mintable amount.
+   */
+  function getMintableAmount() public view returns (uint256) {
+    (uint256 targetGoldTotalSupply, , ) = getTargetGoldTotalSupply();
+    return targetGoldTotalSupply - getGoldToken().totalSupply();
+  }
+
+  /**
+   * @notice Returns the target CELO supply according to the target schedule.
+   * @return targetGoldTotalSupply The target total CELO supply according to the target schedule.
+   * @return communityTargetRewards The community reward that can be minted according to the target schedule.
+   * @return carbonFundTargetRewards The carbon offsetting reward that can be minted according to the target schedule.
+   */
+  function getTargetGoldTotalSupply()
```

Comment:
how should i review this? same logic is in epoch rewards?

Reply:
For the most part, same logic. 

The only difference, is that it's not minting the fraction that would go to validators. This results in less CELO being minted in the first 15 year period than originally planned for.